synapse/events/utils.py

```diff
@@ -506,19 +515,24 @@ def _inject_bundled_aggregations(
                 "sender": edit.sender,
             }
 
-        # If this event is the start of a thread, include a summary of the replies.
-        if aggregations.thread:
-            thread = aggregations.thread
+        # Include any threaded replies to this event.
+        #
+        # Note that is it not valid to start a thread on an event which already
+        # has a relation. Doing so would cause an infinite loop.
+        event_relation = event.content.get("m.relates_to")
+        include_thread = (
+            not isinstance(event_relation, dict)
+            or event_relation.get("rel_type") is None
+        )
```

Comment:
Maybe it would make more sense to pass this in as a flag, like is done for `apply_edits`?